### ISSUE TITLE
Operator 'in' doesn't work with implicit type with index.

### ISSUE THREAD

**Issue by NN---** [Microsoft/TypeScript#1819]
When you write this you get an error:

```typescript
	var q: { [index: number]: Object; } = {};
	if (1 in q) { // The right-hand side of 'in' expression must be of type 'any', an object type of a type parameter.
	}
```

But when you define interface, everything works:

```typescript
	interface Q { [index: number]: Object; };
	var q: Q = {};
	if (1 in q) { // OK
	}
```

TypeScript 1.4

**Comment by RyanCavanaugh** [comment]
This doesn't give an error in 1.4. Can you double-check your code?

**Comment by NN---** [comment]
I have uploaded the full solution here:
https://mega.co.nz/#F!FhcUlAaC!al7uCEPQ8rY2MU8zHYLEWg

**Comment by RyanCavanaugh** [comment]
I don't see an error when opening that solution.

If you have another file open in VS that has a top-level variable called `a` with a different type, you might see this error. Is that happening? Can you post a screenshot maybe?

**Comment by NN---** [comment]
![untitled](https://cloud.githubusercontent.com/assets/580536/5945722/3d1a8264-a739-11e4-879d-310e52e4241c.png)

**Comment by danquirk** [comment]
Strange, no error in the Playground: http://www.typescriptlang.org/Playground#src=var%20uniqueName%3A%20%7B%20%5Bindex%3A%20string%5D%3A%20any%3B%20%7D%20%3D%20%7B%7D%3B%0Aif(1%20in%20uniqueName)%20%7B%0A%7D

**Comment by NN---** [comment]
I know, tried it there but still have problem in my solution.
I have installed "TypeScript Tools for Microsoft Visual Studio 2013 1.4.0.0" .

**Comment by RyanCavanaugh** [comment]
This is very strange. Have you modified lib.d.ts at all?

Can you try this code as well and see if it reports an error?
```ts
var x: {};
if(x instanceof HTMLElement) {}
```

**Comment by mhegazy** [comment]
@NN--- do you get a the same error when you build using the commandline compiler?

**Comment by NN---** [comment]
@RyanCavanaugh @mhegazy Thank you all.
The problem has been found it is a Resharper's bug :)
